Source/FakeItEasy/Core/FakeWrapperConfigurator.cs

```diff
@@ -47,6 +47,13 @@ public IFakeOptionsBuilder<T> WithAdditionalAttributes(IEnumerable<CustomAttribu
 
         public IFakeOptionsBuilder<T> Implements(Type interfaceType)
         {
+            Guard.AgainstNull(interfaceType, "interfaceType");
+
+            if (!interfaceType.IsInterface)
+            {
+                throw new ArgumentException("The specified type is not an interface");
+            }
+

```

Comment:
The change looks good to me, but I'd've thought to just make the adjustment to `FakeOptions.AddInterfaceToImplement`, as that's where all implementations end up.
I'm not necessarily against the way you've done it, but wondered if you had a reason to prefer putting the change in two spots.